pkg/collector/parser/receiver.go

```diff
@@ -155,10 +157,19 @@ func portName(receiverName string, port int32) string {
 }
 
 func portFromEndpoint(endpoint string) (int32, error) {
-	i := strings.LastIndex(endpoint, ":") + 1
-	part := endpoint[i:]
-	port, err := strconv.ParseInt(part, 10, 32)
-	return int32(port), err
+	u, _ := url.Parse(endpoint)
+
+	host, port, _ := net.SplitHostPort(u.Host)
+	
+	if len(host) > 0 {
+		port, err := strconv.ParseInt(port,10,32)
+		return int32(port), err 
+	} else {
```

Comment:
else is not needed given you have the return above

Reply:
ok, I got your point, however, I'm parsing the port variable in a different way on `else`.